### ISSUE TITLE
Stale config values in cache

### ISSUE THREAD

**Issue by NickHeiner** [Microsoft/vscode-eslint#578]
I recently experienced an issue where vscode-eslint's view of the config did not match the files on disk. I edited the files on disk, but vscode-eslint reported the same old config. When I reloaded the window, the problem disappeared.

The file on disk changed as a result of a git checkout.

My versions:

vscode: bc24f98b5f70467bc689abf41cc5550ca637088e.
eslint: v4.19.1.
vscode-eslint: 1.7.0.

I don't have time to build out an easily reproducible example, but I just wanted to share in case other people are encountering the same issue.

**Comment by dbaeumer** [comment]
@NickHeiner you are talking about a `.eslintrc` file right. I test this but of course it did work for me. The extension is listening to file events provided by VS Code itself. Without a reproducible case very hard to track down.

Will keep it open for some time to see if others hit this as well.

**Comment by NickHeiner** [comment]
Yes, I am talking about a file required indirectly from an .eslintrc. And I
know this is basically unactionable without repro steps, but your plan to
keep this open and see if anyone else hits it is precisely what I was
hoping for. :)

**Comment by NickHeiner** [comment]
This problem is happening intermittently for me again. Is there anything I can do to help debug it, while the error conditions are happening? Otherwise, I'll do the workaround again, which is to restart my window.

**Comment by dbaeumer** [comment]
@NickHeiner you can enable tracing to see if the file event for the change of the `eslintrc` file is sent to the server. To enable tracking do:

```
"eslint.trace.server": "verbose"
```

and then add the trace from the ESLint output channel here.
